### PR TITLE
feat: Add button to duplicate instrument (#25649)

### LINKED ISSUE #25649: Duplicate an Instrument
### Your idea

Add the option to duplicate an Instrument. This should apply all formating and notes ect. directly. 
It would become easy to just tranpose by changing the instrument.

### Problem to be solved

When adding transposed parts i need to copy all Notes to the new Staff. But a lot of the formating neets to be applyed manually. This could be simplied.

### Prior art

_No response_

### Additional context

_No response_

### Checklist

- [X] This request follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting issues
- [X] I have verified that this feature request has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar requests

### PR DESCRIPTION
**Pull request by shnCanos**
Closes #25649

- Add button and signal to InstrumentSettingsPopup.qml
- Hook the button to PartTreeItem::duplicateInstrument
- Create PartTreeItem::duplicateInstrument
- Create NotationParts::duplicatePart
  - Create NotationParts::doDuplicatePart
  - Change Excerpt::cloneStaff, Excerpt::cloneStaff2, Excerpt::cloneSpanner to support unlinked clones
- Add ProjectAudioSettings::duplicateTrackParams (and helpers)

- Add Excerpt::cloneStaff tests
- Add ProjectAudioSettings::duplicateTrackParams tests

Resolves: #25649

# Major changes
- Added a button to duplicate an instrument.
- Excerpt::cloneStaff and Excerpt::cloneStaff2 only supported linked clones. Now they (and all helpers) support unlinked clones.
	- Added tests to test this new behaviour and to make sure that the past behaviour is maintained.
- Added a few functions, along with helpers, to duplicate track parameters.
	- Added tests to test these functions.

# Rationale behind some of the decisions

1. Why is `copyAudioStateForNewPart` passed as a lambda instead of just passing the captured variables as parameters?

So as to not have to add `#include "project/..."` inside `notationparts.cpp`.

We do understand that this might not be the best way of going about it, so if there's a better way of doing this (or if adding the include is acceptable) we can change it.

2. Why change `Excerpt::cloneStaff2` to support unlinked clones despite not using it?

Because it shares some of its functions with `Excerpt::cloneStaff`, which we do use.

While we did test it, it was far from as thoroughly as `Excerpt::cloneStaff`. Despite that, the changes should break no past functionality, so there should be no problem in having them.

# Other Notes
- Although we tested it manually, there are sure to be edge cases we didn't take into account. Thus, if anyone has a highly complex score they wouldn't mind sharing, especially one that's using niche use-cases, we'd appreciate it if you sent it here so we could test it and find bugs.
- We tested VSTs programmatically, but not manually. They *should* work, but we'd appreciate it if anyone vouched for the fact that they actually do.

# Bugs

1. Some of the mixer settings don't get duplicated.

This is due to #33391 so it's likely best to not merge this until that's fixed (we verified that it's caused by that by using other unrelated channels to set the values. You can see more information in [my comment](https://github.com/musescore/MuseScore/issues/33391#issuecomment-4566653154)).

2. Creating an instrument, duplicating it, and then deleting results in:

```
ASSERT failure in void QQuickTreeModelAdaptor1::modelRowsAboutToBeRemoved(const QModelIndex&, int, int): "Consistency test failed", file /path/to/MuseScore/muse/framework/uicomponents/qml/Muse/UiComponents/LegacyTreeView/qquicktreemodeladaptor.cpp, line 669`
```

But the same thing happens when creating a score with an instrument, inserting another one, and deleting the first one so we believe it's not our fault. This was only tested with our changes and we didn't find an issue related to this, so if someone manages to replicate, they can let us know and we'll open an issue.

3. Chord brackets are still linked.

We noticed that currently, chord brackets, specifically, are still linked after duplication (changing one changes the other).

This is easily fixed with the diff:

```diff
diff --git a/src/engraving/dom/chord.cpp b/src/engraving/dom/chord.cpp
index 12fd9b026a..7d8b5809f5 100644
--- a/src/engraving/dom/chord.cpp
+++ b/src/engraving/dom/chord.cpp
@@ -392,7 +392,7 @@ Chord::Chord(const Chord& c, bool link)
 
     for (EngravingItem* e : c.el()) {
         if (e->isChordBracket()) {
-            EngravingItem* clonedChordBracket = e->linkedClone();
+            EngravingItem* clonedChordBracket = link ? e->linkedClone() : e->clone();
             add(clonedChordBracket);
         }
     }
```

But this change is outside the scope of the feature. If need-be, I (@shnCanos) can make a separate PR with this change.

We were unable to find an issue relating to this.

# Video showing the feature

https://github.com/user-attachments/assets/2a97f58d-48ec-417f-b039-cc14b9b0f142

[Link to the score used in the video](https://musescore.com/user/101554/scores/117279)

- [x] I signed the [CLA](https://musescore.org/en/cla) as [shncanos](https://musescore.com/user/118014740) (@shnCanos) and [joaocastrosil](https://musescore.com/user/118017353) (@jvcastroo-ist)
- [x] The title of the PR describes the problem it addresses.
- [ ] Each commit's message describes its purpose and effects, and references the issue it resolves. If changes are extensive, there is a sequence of easily reviewable commits.
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/muse_framework/wiki/CodeGuidelines).
- [x] I understand all aspects of the code I'm contributing and I'm able to explain it if requested.
- [x] The code compiles and runs on my machine, preferably after each commit individually. I have manually tested and verified that my changes fulfil their intended purpose.
- [x] No prior attempts to resolve this problem exist, or if they do, I listed them in my PR description and described how I avoided repeating past mistakes.
- [x] There are no unnecessary changes.
- [x] I created a unit test or vtest to verify the changes I made (if applicable).

### CODE REVIEW

**Comment by shnCanos** [comment]
The tests failing doesn't seem to be our fault. We were unable to reproduce this error locally, but it seems it was fixed [here](https://github.com/musescore/muse_framework/commit/e926c9bb38d8e3b3fa4d8990fc29d3a6651502c9#diff-ac1cb7e3f93461782200389a4a0a859889a0536218f3758b17b5b3fe9852eb56R60) and that [other prs](https://github.com/musescore/MuseScore/actions/runs/27061854173/job/79875886111?pr=33658#step:11:510) also had the same error so we'll ignore it for now. Once we rebase to master it should fix itself.

**Comment by jvcastroo-ist** [comment]
Hi @igorkorsukov ,

Just a friendly follow-up on this PR, as it's been a little over two weeks since we opened it. I understand maintainers are busy, but I'd appreciate it if you could take a look whenever you have the chance. If there are any changes you'd like us to make, we'll be happy to update the PR.

Thank you!

**Comment by igorkorsukov** [comment]
> Hi @igorkorsukov ,
> 
> Just a friendly follow-up on this PR, as it's been a little over two weeks since we opened it. I understand maintainers are busy, but I'd appreciate it if you could take a look whenever you have the chance. If there are any changes you'd like us to make, we'll be happy to update the PR.
> 
> Thank you!

My colleagues have taken over this pull request for review. Thank you!
